x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/perform_rule_upgrade/diffable_rule_fields_mappings.test.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformDiffableFieldValues } from './diffable_rule_fields_mappings';
+
+describe('transformDiffableFieldValues', () => {
+  it('transforms rule_schedule into "from" value', () => {
```

Comment:
I think we will need to cover this functionality with API-level tests. The perform upgrade endpoint is quite complex, and the functions it consists of aren’t well-defined in terms of responsibility and encapsulation. This makes it less suited for unit test coverage. Ideally, we should have an integration test for every upgradeable rule field, but that is beyond the scope of this PR.

Reply:
Yeah I agree, I've been adding scenarios like these to the upcoming test plans for FTR and cypress coverage. This was mostly just something to cover it in the meantime.